addons/project/static/src/components/project_right_side_panel/project_right_side_panel.js

```diff
@@ -131,13 +131,22 @@ export class ProjectRightSidePanel extends Component {
     }
 
     async onProjectActionClick(params) {
-        this.actionService.doActionButton({
-            type: 'action',
-            resId: this.projectId,
-            context: this.context,
-            resModel: 'project.project',
-            ...params,
-        });
+        if (params.resId) {
+            this.actionService.doAction({
+                type: "ir.actions.act_window",
+                views: [[false, "form"]],
+                res_id: params.resId,
+                res_model: params.res_model,
+            });
+        } else {
+            this.actionService.doActionButton({
+                type: 'action',
+                resId: this.projectId,
+                context: params.context,
+                resModel: 'project.project',
+                ...params,
+            });
+        }
```

Comment:
I am wondering if we cannot do something like that:
```diff
diff --git a/addons/sale_project/static/src/components/project_right_side_panel/project_right_side_panel.js b/addons/sale_project/static/src/components/project_right_side_panel/project_right_side_panel.js
index b3788d30540..c2def892a7d 100644
--- a/addons/sale_project/static/src/components/project_right_side_panel/project_right_side_panel.js
+++ b/addons/sale_project/static/src/components/project_right_side_panel/project_right_side_panel.js
@@ -27,6 +27,18 @@ patch(ProjectRightSidePanel.prototype, '@sale_project/components/project_right_s
         }
     },
 
+    async onSaleItemActionClick(params) {
+        if (params.resId && params.type !== 'object') {
+            const action = await this.actionService.loadAction(params.name, params.context || {});
+            this.actionService.doAction({
+                ...action,
+                res_id: params.resId,
+            });
+        } else {
+            this.onProjectActionClick(params);
+        }
+    },
+
     formatValue(value, unit) {
         return unit === 'Hours' ? formatFloatTime(value) : formatFloat(value);
     }
diff --git a/addons/sale_project/static/src/components/project_right_side_panel/project_right_side_panel.xml b/addons/sale_project/static/src/components/project_right_side_panel/project_right_side_panel.xml
index a05219d83ab..91ae7ba9ea9 100644
--- a/addons/sale_project/static/src/components/project_right_side_panel/project_right_side_panel.xml
+++ b/addons/sale_project/static/src/components/project_right_side_panel/project_right_side_panel.xml
@@ -24,7 +24,7 @@
                             <t t-set="uom_name" t-value="sale_item.product_uom and sale_item.product_uom[1]"/>
                             <td>
                                 <t t-set="sol_name" t-value="sale_item.display_name"/>
-                                <a t-if="sale_item.action" class="o_rightpanel_button" href="#" t-on-click="() => this.onProjectActionClick(sale_item.action)">
+                                <a t-if="sale_item.action" class="o_rightpanel_button" href="#" t-on-click="() => this.onSaleItemActionClick(sale_item.action)">
                                     <t t-esc="sol_name"/>
                                 </a>
                                 <t t-else="" t-esc="sol_name"/>
```

(I didn't test my suggestion)